ecs-agent/acs/session/task_manifest_common.go

```diff
@@ -0,0 +1,73 @@
+package session
+
+import (
+	apitaskstatus "github.com/aws/amazon-ecs-agent/agent/api/task/status"
+	"github.com/aws/amazon-ecs-agent/agent/logger"
+	"github.com/aws/aws-sdk-go/aws"
+
+	"github.com/aws/amazon-ecs-agent/ecs-agent/acs/model/ecsacs"
+)
+
+// SequenceNumberAccessor stores the latest sequence number that corresponds to the
+// most recently processed task manifest message and is used to determine if a
+// message is stale.
+type SequenceNumberAccessor interface {
+	GetLatestSequenceNumber() int64
+	SetLatestSequenceNumber(sequenceNumber int64)
+}
+
+// ManifestMessageIDAccessor stores the latest message ID that corresponds to the
+// most recently processed task manifest message and is used to determine the
+// validity of a task stop verification ack message.
+type ManifestMessageIDAccessor interface {
+	GetMessageID() string
+	SetMessageID(messageID string) error
+}
+
+type TaskGetter interface {
+	GetRunningTasksOnInstance() ([]string, error)
+}
+
+type TaskStopper interface {
+	StopTask(taskARN string)
+}
+
```

Comment:
Other than the `ManifestMessageIDAccessor` interface, will these interfaces need to be used in both handling / responding to task manifest and task stop verification ACK messages from ACS? I was under the impression that `ManifestMessageIDAccessor` is the only interface that will be common to both, while `SequenceNumberAccessor` and `TaskGetter` are only needed for the former and `TaskStopper` is only needed for the latter.

Reply:
You're right. I just placed them all in this one file but I can place them accordingly in two files: `task_manifest_common.go` and` task_verification_ack_common.go`, would that address your point about the separations?

Reply:
I had imagined there to be 3 files:

1. File for functionality common to handling task manifest and task stop verification ACK messages from ACS contains`ManifestMessageIDAccessor` interface.

2. File for functionality specific to handling task manifest messages from ACS contains `SequenceNumberAccessor` and `TaskGetter` interfaces.

3. File for functionality specific to handling task manifest messages from ACS contains  `TaskStopper` interface.

The first file would be in the scope of this PR and the other two files could follow in subsequent PRs.